vcd/resource_vcd_dnat.go

```diff
@@ -68,6 +73,7 @@ func resourceVcdDNAT() *schema.Resource {
 				Default:  "any",
 			},
 		},
+		DeprecationMessage: "This resource will require network_name in the next major version",
```

Comment:
Does this mean the user will get this message even if he specifie `network_name`? Is there a way to make it smarter, so it only shows up when it's not defined?

Reply:
It will be showed always. We are not deprecate but add property. I don't know other way.

Reply:
This is the wrong way of using deprecation.
The field `DeprecationMessage`should be added to a resource that we leave unchanged, and we use the message to suggest using the new resource.
However, if we modify the resource without creating a new one, the deprecation message does not have sense and should be removed.

Terraform can deprecate either a resource or a property, but not the lack of a property.

Reply:
Well - this is discussion matter, there was intention to have message. If there isn't right way - so we can tell only in documentation, or have message which isn't used properly.

Reply:
Given that Terraform doesn't provide a way to deprecate the lack of property, we should print a warning message on our own instead during apply.

Reply:
Removed this one and added simple message when network name is not provided.